src/workerd/jsg/ser.h

```diff
@@ -33,20 +98,41 @@ class Serializer final: v8::ValueSerializer::Delegate {
 
   KJ_DISALLOW_COPY_AND_MOVE(Serializer);
 
+  kj::Maybe<ExternalHandler&> getExternalHandler() { return externalHandler; }
+
+  // Write a value.
+  //
+  // You can call this multiple times to write multiple values, then call `readValue()` the same
+  // number of times on the deserialization side.
   void write(Lock& js, const JsValue& value);
 
+  // Implements the `transfer` option of `structuredClone()`. Pass each item in the transfer array
+  // to this method before calling `write()`. This gives the Serializer permission to serialize
+  // these values by detaching them (destroying the caller's handle) rather than make a copy. The
+  // detached content will show up as part of `Released`, where it should then be delivered to the
+  // Deserializer later.
   void transfer(Lock& js, const JsValue& value);
 
   Released release();
 
+  void writeRawUint32(uint32_t i) { ser.WriteUint32(i); }
+  void writeRawUint64(uint64_t i) { ser.WriteUint64(i); }
+
+  void writeRawBytes(kj::ArrayPtr<const kj::byte> bytes) {
+    ser.WriteRawBytes(bytes.begin(), bytes.size());
+  }
```

Comment:
non-blocking nit: as a convenience, it may be nice to have this automatically write the size as a prefix.

Also, a `writeString(kj::StringPtr str)` variation would probably be helpful at some point. Both can be done later tho.

Reply:
I don't think it should write the size as a prefix. You might be writing a fixed-size array, or something where the size is known based on some other property read before the bytes.

We could add a separate `writeLengthDelimited()` that writes a length first. (But these can easily be added as needed, as you say.)